source/slang/source-loc.cpp

```diff
@@ -292,6 +292,28 @@ void SourceManager::initialize(
     m_nextLoc = m_startLoc;
 }
 
+UnownedStringSlice SourceManager::allocateStringSlice(const UnownedStringSlice& slice)
+{
+    const UInt numChars = slice.size();
+
+    char* dst = (char*)m_memoryArena.allocate(numChars);
+    ::memcpy(dst, slice.begin(), numChars);
+
+    return UnownedStringSlice(dst, numChars);
+}
+
+UnownedStringSlice SourceManager::allocateConcatStringSlice(const UnownedStringSlice& a, const UnownedStringSlice& b)
+{
```

Comment:
If this is only being used for the `?:` case, then we should be able to eliminate this function.